pkg/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.builds

```diff
@@ -1,12 +1,22 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <TargetRuntimePackageLocation Condition="'$(OSGroup)' == 'Windows_NT'">win</TargetRuntimePackageLocation>
+    <TargetRuntimePackageLocation Condition="'$(OSGroup)' == 'Unix'">linux</TargetRuntimePackageLocation>
+    <TargetRuntimePackageLocation Condition="'$(OSGroup)' == 'OSX'">osx</TargetRuntimePackageLocation>
+  </PropertyGroup>
+  
+  
+
   <ItemGroup>
     <Project Include="Microsoft.DotNet.ILCompiler.pkgproj" >
-      <OSGroup>Windows_NT</OSGroup>
-      <OSGroup>Linux</OSGroup>
-      <OSGroup>Mac</OSGroup>
+      <OSGroup>AnyOS</OSGroup>
     </Project>
+
+    <!-- Include platform-specific package definitions-->
+    <Project Include="$(TargetRuntimePackageLocation)\Microsoft.DotNet.ILCompiler.pkgproj" />
```

Comment:
Are these individual .pkgproj in subdirectories still required now that you do not include them from the metapackage?

I do not see them where we do similar things in other repos. E.g. https://github.com/dotnet/corefx/tree/master/pkg/Microsoft.Private.CoreFx.NETCoreApp .

Reply:
Yes. Removing them causes the runtime packages to not be built. CoreFX's packaging pipeline is much more involved and a lot of the package building for specific platforms is handled outside of the pkproj (most of the logic seems to be here: https://github.com/dotnet/corefx/blob/master/pkg/frameworkPackage.targets). In CoreRT I think this adds some unnecessary complexity, particularly because the package consists entirely of components built by the standard build process and everything else is handled by msbuild. 

I based the subdirectory structures on guidelines here 
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/package-projects.md#platform-specific-library

Reply:
> I based the subdirectory structure on guidelines here
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/package-projects.md#platform-specific-library

This document is outdated. We are not using this style of OS-specific packaging anywhere anymore.

If you replace TargetRuntimePackageLocation with say "TargetSpecific" and have just one unified .pkgproj shared by all OSes/architectures in "TargetSpecific" subdirectory, is it going to work? Or do the build scripts have a strong dependency on the win/linux/osx directory names>.

I would like to minimize the amount of plumbing that needs to be done per new OS target.

Reply:
Seems like the requirements are on a separate `.pkgproj` file with the same name as the lineup package, which has its RID explicitly set. 
Now adding a new OS target should just involve adding a line to the pkproj file.